### PR TITLE
[spirv] emit not preprocessed source code for `DebugSource`

### PR DESCRIPTION
**Pull request by jaebaek**
This commit emits not preprocessed source code for `OpSource`
and `DebugSource`. Now it removes `#line` directives.

### CODE REVIEW

File: tools/clang/lib/SPIRV/EmitVisitor.cpp [hunk 1]
```diff
@@ -96,6 +100,30 @@ bool isBufferBlockDecorationDeprecated(
   return opts.targetEnv.compare("vulkan1.2") >= 0;
 }
 
+// Read the file in |filePath| and returns its contents as a string.
+// This function will be used by DebugSource to get its source code.
+std::string ReadSourceCode(llvm::StringRef filePath) {
+  try {
+    dxc::DxcDllSupport dllSupport;
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/EmitVisitor.cpp:108]: our fork of DXC currently doesn't have the Linux buildbots. Does this code compile on Linux?

**Review by ehsannas — COMMENTED** [review]
It is OK to change the behavior of rich debug info for emitting (or not emitting) source. But it may not be ok to change the behavior for the `-fspv-debug=source` and `-Zi`. I know many devs use `-Zi`. We should check internally to see whether this is ok.

**Comment by jaebaek** [comment]
> It is OK to change the behavior of rich debug info for emitting (or not emitting) source. But it may not be ok to change the behavior for the `-fspv-debug=source` and `-Zi`. I know many devs use `-Zi`. We should check internally to see whether this is ok.

The main change is that this commit lets DXC emit **not preprocessed** source code while the existing DXC emits the preprocessed one. Do you think it still matters? Let me ask Hai about this issue.

File: tools/clang/lib/SPIRV/EmitVisitor.cpp (hunk 1 above)
Comment (jaebaek) [reply tools/clang/lib/SPIRV/EmitVisitor.cpp:108]: Linux build works fine. I made build errors on Windows. I fixed the build failure. PTAL!

**Comment by jaebaek** [comment]
Just for note, we had an offline discuss about the not preprocessed source code and concluded that it is fine.

File: tools/clang/unittests/SPIRV/CodeGenSpirvTest.cpp [hunk 2]
```diff
@@ -1578,10 +1578,6 @@ TEST_F(FileTest, SpirvDebugControlFile) {
   useVulkan1p1();
   runFileTest("spirv.debug.ctrl.file.hlsl");
 }
-TEST_F(FileTest, SpirvDebugControlSource) {
```
Comment (ehsannas) [inline tools/clang/unittests/SPIRV/CodeGenSpirvTest.cpp:1581]: Why is this test removed?

File: tools/clang/test/CodeGenSPIRV/spirv.debug.ctrl.line.hlsl [hunk 3]
```diff
@@ -6,8 +6,6 @@
 // CHECK:      OpSource HLSL 610 [[file]]
 // Have source code
 // CHECK:      float4 main(uint val
-// No tool
-// CHECK-NOT:  OpModuleProcessed
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/spirv.debug.ctrl.line.hlsl:10]: why is this removed?

**Review by ehsannas — CHANGES_REQUESTED** [review]
Please use WholeFileTest fixture so that we don't lose test coverage. Thanks

**Comment by jaebaek** [comment]
As we discussed in offline, I implemented emitting multiple `OpSource`. PTAL!

**Review by ehsannas — APPROVED** [review]
LGTM

**Pull request merged**